rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -499,6 +542,40 @@ mod tests {
         assert_eq!(9, c.value(4));
     }
 
+    #[test]
+    fn test_cast_i32_to_list_i32() {
+        let a = Int32Array::from(vec![5, 6, 7, 8, 9]);
+        let array = Arc::new(a) as ArrayRef;
+        let b = cast(&array, &DataType::List(Box::new(DataType::Int32))).unwrap();
```

Comment:
can we also test length, offset for each value in the casted list array? same in `test_cast_i32_to_list_i32_nullable`.

Reply:
Done for all tests